### DISCUSSION TITLE
What about add searchPattern and searchPatternFlags to restrict search input in Select/MultiSelect?

### DISCUSSION THREAD

**Discussion by dipiash** [mantinedev/mantine#2480] [Feature requests]
Hello!

In our team we are using MultiSelect with live search and we need to restrict search input with reg exp pattern. What about idea to add new props for that in components Select/MultiSelect?

New props:
```tsx
  /** Restrict search input by RegExp */
  searchPattern?: string;

  /** Restrict search input by RegExp flags */
  searchPatternFlags?: string;
```

In `handleSearchChange` (Select/MultiSelect):
```tsx
let preparedVal = val;

if (searchPattern) {
    preparedVal = preparedVal.replace(new RegExp(searchPattern, searchPatternFlags), '');
}
```

I have some example with implementation: e9ae09066e9f850153de8ca0ccd71686bdaf5d44, e9ae09066e9f850153de8ca0ccd71686bdaf5d44 + e9ae09066e9f850153de8ca0ccd71686bdaf5d44 and can create a PR

**Comment by rtivital** [comment]
Hi @dipiash, this does not seem to be a common feature, I'd avoid adding it – components are already very complex

**Reply by dipiash** [reply]
Ok, it seems like that. What about idea to add `searchValue` prop, to make controlled search value from outside of component?

Or maybe you have any ideas how to solve this case?

**Reply by dipiash** [reply]
Hi @rtivital! I created example with controlled/uncontrolled searchValue prop, without new logic, like in my first example:
- MultiSelect - a7331401253a06ecb07341157611848a3ad8dc30
- Select - 9adc817a587b95dc22a1dbd98ae0cd28bf2a6b26

What do you think about it?

**Reply by rtivital** [reply]
I'm fine with this, if you check all Select/MultiSelect demos and confirm that they work as before

**Reply by dipiash** [reply]
Thank you.

Yes, it's works like before. I'll create a PR later today

**Reply by dipiash** [reply]
PR https://github.com/mantinedev/mantine/pull/2485
